java/src/main/java/com/genexus/db/DefaultExceptionErrorHandler.java

```diff
@@ -37,8 +37,17 @@ public static void handleSQLError(IErrorHandler errorHandler, SQLException e, Mo
 					context.globals.Gx_dbsqlstate = e.getSQLState();
 
 					context.inErrorHandler = true;
-					errorHandler.handleError();
-					context.inErrorHandler = false;
+
+					try {
+						errorHandler.handleError();
+					}
+					catch (ReturnInErrorHandlerException e1) {
+						context.inErrorHandler = false;
+						throw e1;
+					}
```

Comment:
The `catch` block is not necessary. Just with the `finally` is enough to reset the `inErrorHandler` flag.

Reply:
Liked it!